### ISSUE TITLE
AI: Typescript code samples for hermes-2-pro-mistral-7b lead to type errors in my local environment

### ISSUE THREAD

**Issue by brettimus** [cloudflare/cloudflare-docs#15930]
### Existing documentation URL(s)

Code Examples \> Worker on this page:
- https://developers.cloudflare.com/workers-ai/models/hermes-2-pro-mistral-7b/

Traditional function calling section on this page:
- https://developers.cloudflare.com/workers-ai/function-calling/

### What changes are you suggesting?

The issue appears to be downstream, having to do with type inference on the `Ai.run` method, however, I would expect the code from the documentation to not give type errors when I copy paste it into my editor:

<img width="644" alt="image" src="https://github.com/user-attachments/assets/984a4cdc-06fe-4a43-923d-5c2121262d47">
<img width="643" alt="image" src="https://github.com/user-attachments/assets/909143cd-517b-42e2-964a-22cd266a84bf">

The problem appears to be that I need to coerce the inputs to: `BaseAiTextGeneration["inputs"]`, which makes type inference work again

<img width="582" alt="image" src="https://github.com/user-attachments/assets/c1c8b827-c21f-4b5e-9396-06162539c8ed">

### Additional information

I created a minimal example by running `npm create cloudflare@latest`, which used `@2.23.0`. I created a Hello World worker, selected Typescript, installed deps. 

I enabled AI bindings, and then copy-pasted the code samples from the pages linked above to test.

Code is [here](https://github.com/brettimus/cloudflare-ai-hermes-2-example)

***

I need to investigate a bit more, but there seem to be several problems with the workers types for AI. (Where should I file an issue about that?) Type inference on `Ai.run` does not work as expected, and then certain types that describe inputs for the AI models have actually caused errors when I conform to them. But that's a different thing I guess.

**Comment by kodster28** [comment]
Believe this is related to https://github.com/cloudflare/workerd/issues/2181

**Comment by brettimus** [comment]
ach sorry, i should've looked in workerd, that makes sense. will check that issue out

**Comment by brettimus** [comment]
it looks like the underlying issue should be resolved soon (given most recent comment on the thread)

i'm guessing it's better to wait for that fix than add a type coercion to the docs, a la `{ messages } as BaseAiTextGeneration["inputs"]`

**Comment by kodster28** [comment]
> it looks like the underlying issue should be resolved soon (given most recent comment on the thread)
> 
> i'm guessing it's better to wait for that fix than add a type coercion to the docs, a la `{ messages } as BaseAiTextGeneration["inputs"]`

Exactly, yeah. As long as there's a fix incoming soon, that'd be our preference.

**Comment by daisyfaithauma** [comment]
Fixed by the updates done on workers AI models
